added browser test to verify attribution section is collapsed by default and expands on clicking

Malar-Natarajan opened the pull request:
GIVEN: A page URL in the format of {base_url}/books/{book_slug}/pages/{page_slug}
    WHEN: The page is fully loaded
    THEN: The attribution section is collapsed by default
    AND The attribution section opens when clicked

TomWoodward committed:
add aria-label to attribution summary

m1yag1 committed:
added attribution-details data-testid

Commit:
updated review comments

Malar-Natarajan committed:
renamed to click_attribution_link

Malar-Natarajan committed:
Updated steps to validate the urls within attribution

Malar-Natarajan committed:
Updated the test to verify section url in attribution section

Malar-Natarajan committed:
updated react snapshots

Malar-Natarajan committed:
test to validate attribution status

Malar-Natarajan committed:
Merge branch 'master' into test-attribution-toggle

Malar-Natarajan committed:
resolved conflicts

Malar-Natarajan committed:
resolving conflicts

TomWoodward commented on .gitignore:
```diff
@@ -71,3 +71,6 @@ pip-delete-this-directory.txt
 # pytest report
 ui-test.html
+pytest-selenium/tests/test_attribution_pages.py
+pytest-selenium/tests/test_attribution_pom.py
```
why are these here?

TomWoodward commented on .gitignore:
```diff
@@ -71,3 +71,6 @@ pip-delete-this-directory.txt
 ui-test.html
+pytest-selenium/tests/test_attribution_pages.py
+pytest-selenium/tests/test_attribution_pom.py
+.idea/
```
please use your [global gitignore](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) for editor files

Malar-Natarajan replied:
@TomWoodward I removed these from the .gitignore and committed the changes.

m1yag1 commented on pytest-selenium/tests/test_attribution_status.py:
```diff
@@ -0,0 +1,33 @@
+):
+
+    """
+    attribution section is initially collapsed, expands when clicked
```
Based on the style guide the first line should have the short description.

m1yag1 replied:
https://wiki.mozilla.org/QA/Execution/Web_Testing/Docs/Automation/StyleGuide

Malar-Natarajan replied:
@m1yag1 I updated the comments and docstrings

m1yag1 commented on pytest-selenium/tests/test_attribution_status.py:
```diff
@@ -0,0 +1,33 @@
+    """
+    attribution section is initially collapsed, expands when clicked
+
+    GIVEN: A page URL in the format of {base_url}/books/{book_slug}/pages/{page_slug}
```
Are you moving the Given When Then into the docstring?

m1yag1 commented on pytest-selenium/tests/test_attribution_status.py:
```diff
@@ -0,0 +1,33 @@
+    assert attribution.is_open
+
+    attribution.click_attribution_link()
+    # """click on attribution again and verify it closes
```
You are using a docstring here which should only come after a `def` or `class` definition. comments within code use a `# `

Malar-Natarajan replied:
@m1yag1  I updated the comments and docstrings

m1yag1 replied:
What's up with line 26. I still see a comment with `"""` inside the code

Malar-Natarajan replied:
@m1yag1  Thats my bad....I updated this

m1yag1 commented on pytest-selenium/tests/test_attribution_status.py:
```diff
@@ -0,0 +1,33 @@
+    GIVEN: A page URL in the format of {base_url}/books/{book_slug}/pages/{page_slug}
+    WHEN: The page is fully loaded
+    THEN: The attribution section is collapsed by default
+    AND The attribution section opens when clicked
```
Also, not a consistent use of the `:` after the test section designator. should be `AND: `

Malar-Natarajan replied:
Updated

Malar-Natarajan committed:
Removing files I added by mistake

I am removing my local files that I added fere by mistake:
pytest-selenium/tests/test_attribution_pages.py
pytest-selenium/tests/test_attribution_pom.py
.idea/

Malar-Natarajan committed:
Updated docstrings and comments

Malar-Natarajan committed:
Merge branch 'test-attribution-toggle' of https://github.com/openstax/rex-web into test-attribution-toggle

Malar-Natarajan committed:
updated docstrings and comments

m1yag1 commented on pytest-selenium/tests/test_attribution_status.py:
```diff
@@ -0,0 +1,30 @@
+def test_attribution_collapsed_by_default_expands_when_clicked(
+    selenium, base_url, book_slug, page_slug
+):
+
```
No line break here. Capitalize first word in docstring. `Attribution`

Malar-Natarajan replied:
@m1yag1 Updated

Malar-Natarajan committed:
updated line break and capitalized first letter

Malar-Natarajan committed:
updated bad docstring

Malar-Natarajan committed:
Merge branch 'master' of https://github.com/openstax/rex-web into test-attribution-toggle

TomWoodward commented on pytest-selenium/tests/test_attribution_status.py:
```diff
@@ -0,0 +1,28 @@
+    assert attribution.is_open
+
+    attribution.click_attribution_link()
+    """click on attribution again and verify it closes"""
```
shouldn't this be a `#` comment?

Malar-Natarajan replied:
@TomWoodward 
I remember Mike saying, I cannot place a comment above a method/function, instead use docstring 
@m1yag1 is my memory bad?

m1yag1 replied:
Yea, that's the right use of a docstring but I think naming the test appropriately would remove the need for the docstring.

Malar-Natarajan replied:
Ok. So I removed the docstring for the click function and just have a comment to explain the assertion

Malar-Natarajan committed:
updated docstring to comment

Malar-Natarajan committed:
making one test file for attribution
>
 Combibining test c476303 & c476302 in a single test file test_attribution

m1yag1 commented on pytest-selenium/pages/content.py:
```diff
@@ -108,4 +104,4 @@ def section_url(self):
         def click_attribution_link(self):
             self.offscreen_click(self.attribution_link)
-            return self.page.attribution.wait_for_region_to_display()
+            self.page.attribution.wait_for_region_to_display()
```
Why was this return removed? It should return the page so that we have a reference to it in the code.

Malar-Natarajan replied:
@m1yag1 
I was discussing about some random locators with Greg and he suggested that a return is not necessary since we are already in that page

m1yag1 replied:
ok then place a `return` only. you don't need to put anything after.

Malar-Natarajan replied:
@m1yag1 
do you mean just have the click function and remove the whole wait for region code
         def click_attribution_link(self):
             self.offscreen_click(self.attribution_link)

Malar-Natarajan replied:
Mike resolving this conservation to leave the code as such

Malar-Natarajan committed:
Deleting test_attribution_status.py to make one file for attribution tests

Deleting test_attribution_status.py to make one file for attribution tests. c476302 & c476303 should now be in test_attribution.py

TomWoodward commented on pytest-selenium/tests/test_attribution.py:
```diff
@@ -23,5 +23,30 @@ def test_section_url_in_citation_text_shows_url_for_current_page(
+    attribution.click_attribution_link()
+
+    # AND: clicking on attribution again collapses it
+    assert not attribution.is_open
```
this formatting is inconsistent, i would expect it to be 

```python
    # AND: The attribution section opens when clicked
    attribution.click_attribution_link()
    assert attribution.is_open

    # AND: clicking on attribution again collapses it
    attribution.click_attribution_link()
    assert not attribution.is_open
```

Malar-Natarajan committed:
updating comment

TomWoodward committed:
Merge branch 'master' into test-attribution-toggle

Malar-Natarajan committed:
Merge branch 'master' into test-attribution-toggle

Malar-Natarajan committed:
Merge branch 'master' into test-attribution-toggle

Malar-Natarajan committed:
Merge branch 'master' of https://github.com/openstax/rex-web into test-attribution-toggle
